department-of-veterans-affairs/vets-website
# Add signature component

## Summary
Update the 'Review application' page in the Pension Benefits application(21P-527EZ) to include the Signature component

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Sign in via mocked authentication at `http://localhost:3001/sign-in/mocked-auth`
3. Start a new application
4. Fill out the 'Applicant information' page including first name, middle name(optional) and last name
5. Open the 'Save in progress' menu
6. Select 'Review and submit' as the return url
7. Go to the 'Review application' page (/review-and-submit)
8. Verify that the 'Statement of Truth' is now at the end of the form instead of the 'privacy' checkbox
9. Verify that both fields in the 'Statement of Truth' are required
10. Verify that the full name must match the entered applicant's name
11. Enter the veteran's full name and check the 'certification' checkbox
12. Verify that all chapters and pages are validated before submitting the form

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73016)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

## Screenshots
| Before | After | Invalid | Valid |
| -------- | -------- | -------- | -------- |
|![localhost_3001_pension_application_527EZ_review-and-submit (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/b65df053-f7b7-48f2-b63c-657c42089dd4)|![localhost_3001_pension_application_527EZ_review-and-submit](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/0ad33cef-aa5c-4278-a1a7-78f8d46de09d)|![localhost_3001_pension_application_527EZ_review-and-submit (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/19f76465-4448-4e7f-b140-2ffadb785bde)|
![localhost_3001_pension_application_527EZ_review-and-submit (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/42ee9f56-8ed1-4641-84c5-50979003cff0)



## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73016)

### Quality Assurance & Testing

### Error Handling


### Authentication
